packages/frontend/src/renderer/notifications.ts

```diff
@@ -1,46 +1,63 @@
 /* global Notification */
 import { soundTypeToAudio } from '../shared/sounds'
-import electronStore from '../shared/electronStore'
-import history from '../shared/history'
-import { DisplayableMessage } from '@quiet/nectar'
-
-export const createNotification = async ({
-  title,
-  body,
-  data
-}: {
+import { PayloadAction } from '@reduxjs/toolkit'
+import { identity, NotificationsOptions, NotificationsSounds, publicChannels as channels, publicChannels, settings, users } from '@quiet/nectar'
+import { all, call, select } from 'typed-redux-saga'
+import store from './store'
+
+export interface NotificationsData {
   title: string
-  body: string
-  data: any
-}) => {
-  const sound = parseInt(electronStore.get('notificationCenter.user.sound'))
-  if (sound) {
-    await soundTypeToAudio[sound].play()
+  message: string
+  sound: NotificationsSounds
+  communityId: string
+  channelName: string
+}
+
+export function* displayMessageNotificationSaga(
+  action: PayloadAction<ReturnType<typeof channels.actions.incomingMessages>['payload']>
+): Generator {
+  const publicChannels = yield* select(channels.selectors.publicChannels)
+  const usersData = yield* select(users.selectors.certificatesMapping)
+  const myIdentity = yield* select(identity.selectors.currentIdentity)
+  const currentChannel = yield* select(channels.selectors.currentChannel)
+  const notificationsOption = yield* select(settings.selectors.getNotificationsOption)
+  const notificationsSound = yield* select(settings.selectors.getNotificationsSound)
+
+  const createNotificationsCalls = action.payload.messages.map((messageData) => {
```

Comment:
Saga must not contain any logic itself. It just delegates the tasks so you should extract it to a separate function and use `yield* call` on it